crates/nix_health/src/check/direnv.rs

```diff
@@ -73,6 +73,20 @@ fn install_check(
     }
 }
 
+/// Verify that direnv was not installed globally, under `/usr`.
+fn is_global(direnv_install: &direnv::DirenvInstall) -> CheckResult {
+    let usr_path = std::path::Path::new("/usr");
+    if direnv_install.bin_path.starts_with(usr_path) {
```

Comment:
A better approach would be to resolve the binary symlink and check if it is in nix store (`/nix/store`):

```
❯ readlink $(which direnv)
/nix/store/birqvvy5rw3jkpy7n59nk1yk8wr3l217-home-manager-path/bin/direnv
```

This will then automatically handle the case of direnv being installed else, like `/opt`, on some systems.

Reply:
`fn is_path_under_nix_store(path: ...) -> bool`

Reply:
> A better approach would be to resolve the binary symlink 

added `canonical_path` field in `DirenvInstall` that stores that path without any symbolic links